parastell/magnet_coils.py

```diff
@@ -401,164 +386,110 @@ class MagnetCoil(object):
-
-        # Project position vector onto cross-section
-        pos_proj = normalize(pos - np.inner(pos, norm) * norm)
-
-        # Compute angle by which to rotate cross-section such that it faces the
-        # origin
-        rot_ang_orig = np.arccos(np.inner(pos_proj, t_vec_rot))
-
-        # Re-orient rotated cross-section such that thickness vector faces
-        # origin
-        cubit.cmd(
-            f"rotate Surface {surf_id} angle {np.rad2deg(rot_ang_orig)} about "
-            "origin 0 0 0 direction " + " ".join(str(i) for i in norm)
-        )
+        self.coords = coords
+        self.tangents = tangents
+        self.center_of_mass = center_of_mass
+        self.width = width
+        self.thickness = thickness
 
     def create_magnet(self):
-        """Creates magnet coil volumes in cubit.
+        """Creates a single magnet coil CAD solid in CadQuery.
 
         Returns:
-            volume_id (int): magnet volume ids in cubit
+            coil (object): cq.Solid object representing a single magnet coil.
         """
-        # Cross-section inititally populated with thickness vector
-        # oriented along x axis
-        t_vec = np.array([1, 0, 0])
+        tangent_vectors = [
+            cq.Vector(tuple(tangent)) for tangent in self.tangents
+        ]
 
-        # Create cross-section for sweep
-        cubit.cmd(f"create surface " + self.shape_str + " zplane")
+        # Define coil filament path normals such that they face the filament
+        # center of mass
+        normal_dirs = np.array([i - self.center_of_mass for i in self.coords])
+        normal_dirs = (
+            normal_dirs / np.linalg.norm(normal_dirs, axis=1)[:, np.newaxis]
+        )
 
-        # Store cross-section index
-        cs_id = cubit.get_last_id("surface")
-        # Cross-section initially populated with normal oriented along z
-        # axis
-        cs_axis = np.array([0, 0, 1])
+        # Project normal directions onto desired coil cross-section (CS) plane
+        # at each filament position to define true filament path normals
+        parallel_parts = []
+        for dir, tangent in zip(normal_dirs, self.tangents):
+            parallel_parts.append(np.dot(dir, tangent))
+        parallel_parts = np.array(parallel_parts)
 
-        # Initialize path list
-        path = []
+        normals = normal_dirs - parallel_parts[:, np.newaxis] * self.tangents
+        normals = normals / np.linalg.norm(normals, axis=1)[:, np.newaxis]
 
-        # Create vertices in filament path
-        for x, y, z in self.filament:
-            cubit.cmd(f"create vertex {x} {y} {z}")
-            path += [cubit.get_last_id("vertex")]
+        # Compute binormals projected onto CS plane at each position
+        binormals = np.cross(self.tangents, normals)
 
-        # Ensure final vertex in path is the same as the first
-        path += [path[0]]
+        # Compute coordinates of edges of rectangular coils
+        coil_edge_coords = []
```

Comment:
This can be done in a loop with something like:
```
edge_offsets = np.array([ [-1,-1], [1,-1], [1,1], [-1, 1] ])

for edge_offset in edge_offsets:
    coil_edge = self.coords + 
                edge_offset[0] * binormals * (self.width /2) +
                edge_offset[1] * binormals * (self.thickness /2)

   coil_edge_coords.append(
        [cq.Vector(tuple(pos)) for pos in coil_edge]
        )

```